substrate/frame/support/src/genesis_builder_helper.rs

```diff
@@ -55,3 +55,82 @@ where
 		preset_for_name,
 	)
 }
+
+/// Creates a `RuntimeGenesisConfig` JSON patch.
+///
+/// This macro creates a default `RuntimeGenesisConfig` initializing provided fields with given
+/// values, serialize it to JSON blob, and retain only the specified fields.
+///
+/// This macro helps to prevent errors that could occur from manually creating JSON objects, such
+/// as typos or discrepancies caused by future changes to the `RuntimeGenesisConfig` structure. By
+/// using the actual struct, it ensures that the JSON generated is valid and up-to-date.
+///
+/// This macro assumes that `serde(rename_all = "camelCase")` attribute is used for
+/// RuntimeGenesisConfig, what should be the case for frame-based runtimes.
+///
+/// # Example
+///
+/// ```rust
+/// use frame_support::runtime_genesis_config_json;
+/// #[derive(Default, serde::Serialize, serde::Deserialize)]
+/// #[serde(rename_all = "camelCase")]
+/// struct RuntimeGenesisConfig {
+///     a_field: u32,
+///     b_field: u32,
+///     c_field: u32,
+/// }
+/// assert_eq!(
+///     runtime_genesis_config_json! ({a_field:31, b_field:127}),
+///     serde_json::json!({"aField":31, "bField":127})
+/// );
+/// assert_eq!(
+///     runtime_genesis_config_json! ({a_field:31}),
+///     serde_json::json!({"aField":31})
+/// );
+/// ```
+#[macro_export]
+macro_rules! runtime_genesis_config_json {
```

Comment:
But what about sub maps? Basically like:

```
pallet_whatever: { 
   i_only_want_to_provide_this_field: "cool",
}
```

Reply:
yeah, this proposal does not cover this. support for recursive partial structs would a bit more complex.

Reply:
Then we should support this as well.